client/landing/subscriptions/components/site-subscription-page/site-subscription-page.tsx

```diff
@@ -140,59 +39,23 @@ const SiteSubscriptionPage = () => {
 
 			<div className="site-subscription-page__centered-content">
 				<div className="site-subscription-page__main-content">
-					<header className="site-subscription-page__header site-subscription-page__centered-content">
-						<SiteIcon iconUrl={ data?.site_icon } size={ 116 } />
-						<FormattedHeader brandFont headerText={ data?.name } subHeaderText={ subHeaderText } />
-					</header>
-
-					<Notice
-						onClose={ notice?.onClose }
-						visible={ !! notice }
-						type={ notice?.type }
-						action={ notice?.action }
-					>
-						{ notice?.message }
-					</Notice>
-
-					{ siteSubscribed && (
-						<>
-							<SiteSubscriptionSettings
-								blogId={ blogId }
-								notifyMeOfNewPosts={ data.delivery_methods?.notification?.send_posts }
-								emailMeNewPosts={ data.delivery_methods?.email?.send_posts }
-								deliveryFrequency={ data.delivery_methods?.email?.post_delivery_frequency }
-								emailMeNewComments={ data.delivery_methods?.email?.send_comments }
-							/>
-
-							<hr className="subscriptions__separator" />
-
-							{ /* TODO: Move to SiteSubscriptionInfo component when payment details are in. */ }
-							<div className="site-subscription-info">
-								<h2 className="site-subscription-info__heading">{ translate( 'Subscription' ) }</h2>
-								<dl className="site-subscription-info__list">
-									<dt>{ translate( 'Date' ) }</dt>
-									<dd>
-										<TimeSince
-											date={
-												( date_subscribed?.valueOf()
-													? date_subscribed
-													: new Date( 0 )
-												).toISOString?.() ?? date_subscribed
-											}
-										/>
-									</dd>
-								</dl>
-							</div>
-
-							<Button
-								className="site-subscription-page__unsubscribe-button"
-								isSecondary
-								onClick={ () => unsubscribe( { blog_id: blogId, url: data.URL } ) }
-								disabled={ unsubscribing }
-							>
-								{ translate( 'Cancel subscription' ) }
-							</Button>
-						</>
+					{ error || ! blogId || ! data || isSiteSubscriptionDetailsErrorResponse( data ) ? (
+						<Notice
+							className="site-subscription-page__fetch-details-error"
+							type={ NoticeType.Error }
+						>
+							Subscription not found
+						</Notice>
+					) : (
+						<SiteSubscriptionDetails
+							name={ data.name }
+							subscriberCount={ data.subscriber_count }
+							dateSubscribed={ data.date_subscribed }
+							siteIcon={ data.site_icon }
+							blogId={ data.blog_ID }
```

Comment:
This should help:

```suggestion
							blogId={ blogId }
```

Reply:
Thanks @ivan-ottinger so much for spotting this, and debugging it. It helped me to save some time. 🙇  You are right that the usage of the `data.blog_ID` was the issue here.  For internal users, the `data.blog_ID` type is `number`, while the `blogId` is string. Using blog ID's of different types in react-query key is problematic.

Reply:
The issue should be resolved with my latest commit. 🙇